vllm/model_executor/layers/attention.py

```diff
@@ -156,7 +156,8 @@ def single_query_cached_kv_attention(
         # sequences or heads is large, we use V1 since there is enough work
         # to parallelize.
         # TODO(woosuk): Tune this heuristic.
-        use_v1 = max_num_partitions == 1 or num_seqs * num_heads > 512
+        use_v1 = input_metadata.max_context_len <= 8192 and (
```

Comment:
Let's add a comment here?
```suggestion
        # For context len > 8192, use V2 kernel to avoid shared memory shortage.
        use_v1 = input_metadata.max_context_len <= 8192 and (
```